src/pipelinedefinition.cpp

```diff
@@ -469,16 +469,17 @@ class NodeValidator {
         // If validated connection pair connects two DL model/Custom nodes,
         // check if both input/output exist and its metadata (shape, precision) matches.
         // Affect shape by demultiplexer/gather if applies.
-        const auto& tensorInput = this->inputsInfo.at(modelInputName);
         const auto& tensorOutput = this->dependencyOutputsInfo.at(modelOutputName);
-        shape_t tensorInputShape = tensorInput->getShape();
         shape_t tensorOutputShape = tensorOutput->getShape();
-        if (dependencyNodeInfo.demultiplyCount) {
-            auto result = influenceShapeWithDemultiplexer(tensorOutputShape, dependencyNodeInfo);
-            if (!result.ok()) {
-                return result;
-            }
+        auto result = influenceShapeWithDemultiplexer(tensorOutputShape, dependencyNodeInfo);
+        if (!result.ok()) {
+            return result;
+        }
+        if (dependantNodeInfo.kind == NodeKind::EXIT) {
```

Comment:
This is kind of hack to force demultiplexer demultiplyCount check in `influenceShapeWithDemultiplexer`.
Maybe extract this check out of `influenceShapeWithDemultiplexer` into separate method and call outside of `checkConnectionMetadataCorrectness`?